In [2]:
import pandas as pd

In [3]:
data = pd.read_csv('두레팜-01-2025-10-14-00시~2025-11-13-23시.csv')

In [4]:
data.head()

데이터 수신 시간 운전 상태 운전 모드  재배 일자  CO2 평균값  온도 평균값  습도 평균값  조도 평균값  CO2_1  \
0  2025-10-14 0:00    운전  단일제어     99      369    22.6    90.8       0    369   
1  2025-10-15 0:00    운전  단일제어     99      384    21.1    83.2       0    384   
2  2025-10-16 0:00    운전  단일제어     99      392    21.4    86.1       0    392   
3  2025-10-17 0:00    운전  단일제어     99      379    20.4    81.9       0    379   
4  2025-10-18 0:00    운전  단일제어     99      395    21.4    84.2       0    395   

   온도_1  ...  조도 제어방식  조도 설정값1  조도 설정값2  CO2 릴레이동작  난방 릴레이 동작  냉방 릴레이 동작  \
0  22.6  ...       고정     6000   60분/0분         정지         정지         정지   
1  21.1  ...       고정     6000   60분/0분         정지         동작         정지   
2  21.4  ...       고정     6000   60분/0분         정지         동작         정지   
3  20.4  ...       고정     6000   60분/0분         정지         동작         정지   
4  21.4  ...       고정     6000   60분/0분         정지         동작         정지   

   가습 릴레이 동작  제습 릴레이 동작  조명 릴레이 동작  알람 릴레이 동작  
0         정지         정지         동작         정지  
1         동작         정지         동작         동작  
2         동작         정지         동작         동작  
3         동작         정지         동작         동작  
4         동작         정지         동작         동작  

[5 rows x 43 columns]

In [5]:
data.columns

Index(['데이터 수신 시간', '운전 상태', '운전 모드', '재배 일자', 'CO2 평균값', '온도 평균값', '습도 평균값',
       '조도 평균값', 'CO2_1', '온도_1', '습도_1', '조도_1', 'CO2_2', '온도_2', '습도_2',
       '조도_2', 'CO2_3', '온도_3', '습도_3', '조도_3', 'CO2_4', '온도_4', '습도_4',
       '조도_4', 'CO2 제어방식', 'CO2 설정값1', 'CO2 설정값2', '온도 제어방식', '온도 설정값1',
       '온도 설정값2', '습도 제어방식', '습도 설정값1', '습도 설정값2', '조도 제어방식', '조도 설정값1',
       '조도 설정값2', 'CO2 릴레이동작', '난방 릴레이 동작', '냉방 릴레이 동작', '가습 릴레이 동작',
       '제습 릴레이 동작', '조명 릴레이 동작', '알람 릴레이 동작'],
      dtype='object')

In [50]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31 entries, 0 to 30
Data columns (total 43 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   데이터 수신 시간  31 non-null     object 
 1   운전 상태      31 non-null     object 
 2   운전 모드      31 non-null     object 
 3   재배 일자      31 non-null     int64  
 4   CO2 평균값    31 non-null     int64  
 5   온도 평균값     31 non-null     float64
 6   습도 평균값     31 non-null     float64
 7   조도 평균값     31 non-null     int64  
 8   CO2_1      31 non-null     int64  
 9   온도_1       31 non-null     float64
 10  습도_1       31 non-null     float64
 11  조도_1       31 non-null     int64  
 12  CO2_2      31 non-null     int64  
 13  온도_2       31 non-null     int64  
 14  습도_2       31 non-null     int64  
 15  조도_2       31 non-null     int64  
 16  CO2_3      31 non-null     int64  
 17  온도_3       31 non-null     int64  
 18  습도_3       31 non-null     int64  
 19  조도_3       31 non-null     int64  
 20  CO2_4      3

In [51]:
data.describe()

재배 일자     CO2 평균값     온도 평균값     습도 평균값  조도 평균값       CO2_1       온도_1  \
count   31.0   31.000000  31.000000  31.000000    31.0   31.000000  31.000000   
mean    99.0  375.419355  14.848387  72.832258     0.0  375.419355  14.848387   
std      0.0   10.735530   4.160358   8.241698     0.0   10.735530   4.160358   
min     99.0  349.000000   9.000000  57.400000     0.0  349.000000   9.000000   
25%     99.0  367.000000  11.450000  68.050000     0.0  367.000000  11.450000   
50%     99.0  374.000000  13.100000  72.300000     0.0  374.000000  13.100000   
75%     99.0  384.000000  17.400000  77.700000     0.0  384.000000  17.400000   
max     99.0  395.000000  22.600000  90.800000     0.0  395.000000  22.600000   

            습도_1  조도_1  CO2_2  ...  습도_3  조도_3  CO2_4  온도_4  습도_4  조도_4  \
count  31.000000  31.0   31.0  ...  31.0  31.0   31.0  31.0  31.0  31.0   
mean   72.832258   0.0    0.0  ...   0.0   0.0    0.0   0.0   0.0   0.0   
std     8.241698   0.0    0.0  ...   0.0   0.0    0.0   0.0   0.0   0.0   
min    57.400000   0.0    0.0  ...   0.0   0.0    0.0   0.0   0.0   0.0   
25%    68.050000   0.0    0.0  ...   0.0   0.0    0.0   0.0   0.0   0.0   
50%    72.300000   0.0    0.0  ...   0.0   0.0    0.0   0.0   0.0   0.0   
75%    77.700000   0.0    0.0  ...   0.0   0.0    0.0   0.0   0.0   0.0   
max    90.800000   0.0    0.0  ...   0.0   0.0    0.0   0.0   0.0   0.0   

       CO2 설정값1  온도 설정값1  습도 설정값1  조도 설정값1  
count      31.0     31.0     31.0     31.0  
mean     1400.0     23.0     90.0   6000.0  
std         0.0      0.0      0.0      0.0  
min      1400.0     23.0     90.0   6000.0  
25%      1400.0     23.0     90.0   6000.0  
50%      1400.0     23.0     90.0   6000.0  
75%      1400.0     23.0     90.0   6000.0  
max      1400.0     23.0     90.0   6000.0  

[8 rows x 25 columns]

## 데이터 수신 시간
- 일자별로 존재함.

## 운전 상태
- '운전' 값 단일 존재

## 운전 모드
- '단일제어' 값 단일 존재

## 재배 일자 --> NaN
- 99 값 단일 존재

## CO2 평균값
- 369, 384, 392, 379, 395, 387, 391, 370, 365, 380, 389, 374, 377, 366, 368, 367, 364, 372, 363, 349

## 온도 평균값
- 22.6, 21.1, 21.4, 20.4, 20.2, 19.2, 17. , 17.3, 16.6, 17.1, 17.4, 14.7, 10.8, 10.1, 11. , 12.8, 12.3, 13.1,  9.7, 11.3, 12. , 11.6, 12.6, 15.1, 11.8,  9.1,  9. , 11.1

## 습도 평균값
- 90.8, 83.2, 86.1, 81.9, 84.2, 80.8, 80. , 71.2, 72.4, 73.3, 74.7, 75.2, 75.4, 66.6, 57.4, 57.6, 64. , 71.7, 67.9, 68.3, 60.5, 68.2, 71.9, 69.3, 72.3, 74.3, 83.5, 64.3, 64.5, 75.1

## 조도 평균값
- 0 값 단일 존재

## CO2_1 (위 CO2 평균값과 같음.)
- 369, 384, 392, 379, 395, 387, 391, 370, 365, 380, 389, 374, 377, 366, 368, 367, 364, 372, 363, 349

## 온도_1 (위 온도 평균값과 같음.)
- 22.6, 21.1, 21.4, 20.4, 20.2, 19.2, 17. , 17.3, 16.6, 17.1, 17.4, 14.7, 10.8, 10.1, 11. , 12.8, 12.3, 13.1,  9.7, 11.3, 12. , 11.6, 12.6, 15.1, 11.8,  9.1,  9. , 11.1

## 습도_1
- 90.8, 83.2, 86.1, 81.9, 84.2, 80.8, 80. , 71.2, 72.4, 73.3, 74.7, 75.2, 75.4, 66.6, 57.4, 57.6, 64. , 71.7, 67.9, 68.3, 60.5, 68.2, 71.9, 69.3, 72.3, 74.3, 83.5, 64.3, 64.5, 75.1

## 조도_1
- 0 값 단일 존재

## CO2, 온도, 습도, 조도 (2~4)
- 0 값 단일 존재

## CO2 제어방식
- '자동' 값 단일 존재

## CO2 설정값1
- 1400 값 단일 존재

## CO2 설정값2
- '10분/10분' 값 단일 존재

## 온도 제어방식
- '자동' 값 단일 존재

## 온도 설정값1
- 23 값 단일 존재

## 온도 설정값2
- '10분/10분' 값 단일 존재

## 습도 제어방식
- '자동' 값 단일 존재

## 습도 설정값1
- 90 값 단일 존재

## 습도 설정값2
- '10분/10분' 값 단일 존재

## 조도 제어방식
- '고정' 값 단일 존재

## 조도 설정값1
- 6000 값 단일 존재

## 조도 설정값2
- '60분/0분' 값 단일 존재

## CO2 릴레이동작
- '정지' 값 단일 존재

## 난방 릴레이 동작
- '정지', '동작' 두 개의 값 존재 (0과 1)

## 냉방 릴레이 동작
- '정지' 값 단일 존재

## 가습 릴레이 동작
- '정지', '동작' 두 개의 값 존재 (0과 1)

## 제습 릴레이 동작
- '정지' 값 단일 존재

## 조명 릴레이 동작
- '동작' 값 단일 존재

## 알람 릴레이 동작
- '정지', '동작' 두 개의 값 존재 (0과 1)

In [48]:
data['알람 릴레이 동작'].unique()

array(['정지', '동작'], dtype=object)

In [ ]:
# Index(['데이터 수신 시간', '운전 상태', '운전 모드', '재배 일자', 'CO2 평균값', '온도 평균값', '습도 평균값',
      #  '조도 평균값', 'CO2_1', '온도_1', '습도_1', '조도_1', 'CO2_2', '온도_2', '습도_2',
      #  '조도_2', 'CO2_3', '온도_3', '습도_3', '조도_3', 'CO2_4', '온도_4', '습도_4',
      #  '조도_4', 'CO2 제어방식', 'CO2 설정값1', 'CO2 설정값2', '온도 제어방식', '온도 설정값1',
      #  '온도 설정값2', '습도 제어방식', '습도 설정값1', '습도 설정값2', '조도 제어방식', '조도 설정값1',
      #  '조도 설정값2', 'CO2 릴레이동작', '난방 릴레이 동작', '냉방 릴레이 동작', '가습 릴레이 동작',
      #  '제습 릴레이 동작', '조명 릴레이 동작', '알람 릴레이 동작'],
      # dtype='object')